plugins/catalog-backend/src/processing/TaskPipeline.ts

```diff
@@ -85,34 +89,36 @@ export function startTaskPipeline<T>(options: Options<T>) {
 
   async function pipelineLoop() {
     while (!abortSignal.aborted) {
-      if (state.inFlightCount <= lowWatermark) {
-        const loadCount = highWatermark - state.inFlightCount;
-        const loadedItems = await Promise.resolve()
-          .then(() => loadTasks(loadCount))
-          .catch(() => {
-            // Silently swallow errors and go back to sleep to try again; we
-            // delegate to the loadTasks function itself to catch errors and log
-            // if it so desires
-            return [];
-          });
-        if (loadedItems.length && !abortSignal.aborted) {
-          state.inFlightCount += loadedItems.length;
-          for (const item of loadedItems) {
-            Promise.resolve()
-              .then(() => processTask(item))
-              .catch(() => {
-                // Silently swallow errors and go back to sleep to try again; we
-                // delegate to the processTask function itself to catch errors
-                // and log if it so desires
-              })
-              .finally(() => {
-                state.inFlightCount -= 1;
-                barrier.release();
-              });
+      await withActiveSpan(tracer, 'TaskPipelineLoop', async span => {
```

Comment:
Noting that this makes spans even when there is no work to be done at all (eg even when the watermarks haven't been hit). Is that really intentional? Seems to me you'd get lots of small "junk" traces mixed up with actual work right?

Reply:
As you can see in screenshots in description, without this change there are even more spans, as loadTasks performs dB queries, which are instrumented, but lack proper root context.

Reply:
Sure but it's even outside the if statement that checks watermarks, which does no active work whatsoever.

Reply:
moved inside `if`